### ISSUE TITLE
Integrating with OSS-Fuzz

### ISSUE THREAD

**Issue by Google-Autofuzz** [rougier/freetype-gl#209]
Greetings freetype-gl developers and contributors,

We’re reaching out because your project is an important part of the open source ecosystem, and we’d like to invite you to integrate with our [fuzzing](https://www.owasp.org/index.php/Fuzzing) service, [OSS-Fuzz]( https://opensource.googleblog.com/2016/12/announcing-oss-fuzz-continuous-fuzzing.html ). OSS-Fuzz is a free fuzzing infrastructure you can use to identify security vulnerabilities and stability bugs in your project. OSS-Fuzz will:

- Continuously run all the fuzzers you write.
- Alert you when it finds issues.
- Automatically close issues after they’ve been fixed by a commit.

Many widely used [open source projects]( https://github.com/google/oss-fuzz/tree/master/projects ) like OpenSSL, FFmpeg, LibreOffice, and ImageMagick are fuzzing via OSS-Fuzz, which helps them find and remediate [critical issues]( https://bugs.chromium.org/p/oss-fuzz/issues/list?can=1&q=status%3AFixed%2CVerified+Type%3ABug%2CBug-Security+-component%3AInfra+ ). 

Even though typical integrations can be done in < 100 LoC, we have a [reward program]( https://www.google.com/about/appsecurity/patch-rewards/ ) in place which aims to recognize folks who are not just contributing to open source, but are also working hard to make it more secure.

We want to stress that anyone who meets the eligibility criteria and integrates a project with OSS-Fuzz is eligible for a reward.

If you're not interested in integrating with OSS-Fuzz, it would be helpful for us to understand why—lack of interest, lack of time, or something else—so we can better support projects like yours in the future.

If we’ve missed your question in our [FAQ]( https://github.com/google/oss-fuzz/blob/master/docs/faq.md ), feel free to reply or reach out to us at oss-fuzz-outreach@googlegroups.com.

Thanks!

Julien,
OSS-Fuzz Team

**Comment by rougier** [comment]
Anybody familiar with this service ?

**Comment by Google-Autofuzz** [comment]
If you need help, we can help to start the integration. Do you want us to help?

**Comment by rougier** [comment]
No, I would like first to have more explanation on your first message (is that an automated message, what does the attached code is supposed to explain, etc.). I looked at your FAQ but is is far from clear.

**Comment by Google-Autofuzz** [comment]
Sure, sorry if the meaning wasn't clear.

The first message isn't an automated one, hence why it was signed by Julien, who is a Google employee doing some outreach for [OSS-Fuzz](https://google.github.io/oss-fuzz/).

The attached code is a [fuzzer](https://en.wikipedia.org/wiki/Fuzzing), that we are currently running internally. It feeds random data to functions from the freetype-gl API, trying to uncover problematic outcomes, like hangs or crashes, in order to fix them, to improve the reliability and security of freetype-gl.

OSS-Fuzz is a service that could be compared to appveyor or travis-ci, that you're already using, but instead of trying to simply build your software and run your testsuite, it will continually fuzz it, and let you know if a crash or a hang is found. 

The goal of the first message was to advertise OSS-Fuzz to you, ask if you would be interested in integrating freetype-gl in it, offer monetary compensation for this work, and give you our fuzzer as a starting point.
